File: 0900-rle-iterator/0900-rle-iterator.py
class RLEIterator:

    def __init__(self, encoding: List[int]):
        self.i = 0
        self.j = 0
        self.encoding = encoding

    def next(self, n: int) -> int:
        while self.i < len(self.encoding):
            if n > self.encoding[self.i] - self.j:
                n -= self.encoding[self.i] - self.j
                self.j = 0
                self.i += 2
            else:
                self.j += n
                return self.encoding[self.i+1]
        return -1
        


# Your RLEIterator object will be instantiated and called as such:
# obj = RLEIterator(encoding)
# param_1 = obj.next(n)

File: 0900-rle-iterator/0900-rle-iterator.java
class RLEIterator {
    private int[] encoding;
    private int i = 0;
    private int j = 0;
    public RLEIterator(int[] encoding) {
        this.encoding = encoding;
    }
    
    public int next(int n) {
        while(i < encoding.length) {
            if(n > (encoding[i] - j)) {
                n -= encoding[i] - j;
                j = 0;
                i += 2;
            } else {
                j += n;
                return encoding[i+1];
            }
        }
        return -1;
    }
}

/**
 * Your RLEIterator object will be instantiated and called as such:
 * RLEIterator obj = new RLEIterator(encoding);
 * int param_1 = obj.next(n);
 */